src/api/entitlements.js

```diff
@@ -26,15 +26,21 @@ export class Entitlements {
    * @param {!Array<!Entitlement>} entitlements
    * @param {?string} currentProduct
    * @param {function(!Entitlements)} ackHandler
+   * @param {boolean=} opt_isReadyToPay
    */
-  constructor(service, raw, entitlements, currentProduct, ackHandler) {
+  constructor(
+    service, raw, entitlements, currentProduct, ackHandler, opt_isReadyToPay) {
+
     /** @const {string} */
     this.service = service;
     /** @const {string} */
     this.raw = raw;
     /** @const {!Array<!Entitlement>} */
     this.entitlements = entitlements;
 
+    /** @const {boolean|undefined} */
+    this.isReadyToPay = opt_isReadyToPay;
```

Comment:
Please also pass it in the `clone()` and `json()` methods below.

Reply:
done